spring-boot-project/spring-boot/src/main/java/org/springframework/boot/context/properties/PropertyMapper.java

```diff
@@ -333,6 +334,22 @@ public void to(Consumer<T> consumer) {
 			return factory.apply(value);
 		}
 
+		/**
+		 * Complete the mapping by creating a new instance from the non-filtered value.
+		 * @param <R> the resulting type
+		 * @param factory the factory used to create the instance
+		 * @param defaultValue the default value used in case of predicate is false.
+		 * @return the instance
+		 */
+		public <R> R toInstance(Function<T, R> factory, R defaultValue) {
```

Comment:
I don't think this method is needed. The existing `toInstance(Function<T, R> factory)` method should do what we need when given a method reference to one of `Mustache.Compiler`'s configuration methods that return a new `Compiler` instance.